sycl/source/detail/device_binary_image.cpp

```diff
@@ -235,6 +250,420 @@ DynRTDeviceBinaryImage::~DynRTDeviceBinaryImage() {
+
+    if (!Aspects.empty())
+      Result += strlen("aspects") + 1 + GetAspectsContentSize();
+
+    if (!JointMatrix.empty())
+      Result +=
+          strlen("joint_matrix") + 1 + GetStringSetContentSize(JointMatrix);
+
+    if (!JointMatrixMad.empty())
+      Result += strlen("joint_matrix_mad") + 1 +
+                GetStringSetContentSize(JointMatrixMad);
+
+    return Result;
+  }
+
+  void WriteAspectProperty(sycl_device_binary_property &NextFreeProperty,
+                           char *&NextFreeContent) const {
+    if (Aspects.empty())
+      return;
+    // Get the next free property entry and move the needle.
+    sycl_device_binary_property NewProperty = NextFreeProperty++;
+    NewProperty->Type = SYCL_PROPERTY_TYPE_BYTE_ARRAY;
+    NewProperty->ValSize = GetAspectsContentSize();
+    // Copy the name.
+    const size_t NameLen = std::strlen("aspects");
+    std::memcpy(NextFreeContent, "aspects", NameLen + 1);
+    NewProperty->Name = NextFreeContent;
+    NextFreeContent += NameLen + 1;
+    // Copy the values.
+    uint32_t *AspectContentIt = reinterpret_cast<uint32_t *>(NextFreeContent);
+    for (uint32_t Aspect : Aspects)
+      *(AspectContentIt++) = Aspect;
+    NewProperty->ValAddr = NextFreeContent;
+    NextFreeContent += NewProperty->ValSize;
+  }
+
+  static void WriteStringSetProperty(
+      const std::set<std::string_view> &Set, const char *SetName,
+      sycl_device_binary_property &NextFreeProperty, char *&NextFreeContent) {
+    if (Set.empty())
+      return;
+    // Get the next free property entry and move the needle.
+    sycl_device_binary_property NewProperty = NextFreeProperty++;
+    NewProperty->Type = SYCL_PROPERTY_TYPE_BYTE_ARRAY;
+    NewProperty->ValSize = GetStringSetContentSize(Set);
+    // Copy the name.
+    const size_t NameLen = std::strlen(SetName);
+    std::memcpy(NextFreeContent, SetName, NameLen + 1);
+    NewProperty->Name = NextFreeContent;
+    NextFreeContent += NameLen + 1;
+    // Copy the values.
+    NewProperty->ValAddr = NextFreeContent;
+    for (auto StrIt = Set.begin(); StrIt != Set.end(); ++StrIt) {
+      if (StrIt != Set.begin())
+        *(NextFreeContent++) = ';';
+      std::memcpy(NextFreeContent, StrIt->data(), StrIt->size());
+      NextFreeContent += StrIt->size();
+    }
+  }
+};
+
+// Merging device requirements is a little more involved, as it may impose
+// new requirements.
+static MergedDeviceRequirements
+MergeDeviceRequirements(const std::vector<const RTDeviceBinaryImage *> &Imgs) {
+  MergedDeviceRequirements MergedReqs;
+  for (const RTDeviceBinaryImage *Img : Imgs) {
+    const RTDeviceBinaryImage::PropertyRange &Range =
+        Img->getDeviceRequirements();
+    for (const sycl_device_binary_property Prop : Range) {
+      std::string_view NameView = std::string_view{Prop->Name};
+
+      // Aspects we collect in a set early and add them afterwards.
+      if (NameView == std::string_view{"aspects"}) {
```

Comment:
Can we not just do `NameView == "aspects"` here and below?

Reply:
Sure can!